### PR TITLE
Upgrade to latest TypeScript

### PR DESCRIPTION
**Pull request by robinandeer**
## What

Update to official rollup typescript plugin.

## Notes

The update requires a fix in Storybook so that's why this branch is on top of #579

### CODE REVIEW

File: src/Components/templateExpression.ts [hunk 1]
```diff
@@ -132,10 +132,9 @@ const parseAbstractExpression = (
         }
         if (abstractExpression) {
           if (abstractExpression.type === 'binaryOperator') {
-            abstractExpression = {
-              ...abstractExpression!,
+            abstractExpression = Object.assign(abstractExpression, {
```
Comment (styrelius) [inline src/Components/templateExpression.ts:135]: Why is this changed?

**Review by styrelius — COMMENTED** [review]
Nice! 🏗️ Just a question

File: src/Components/templateExpression.ts (hunk 1 above)
Comment (robinandeer) [reply src/Components/templateExpression.ts:135]: Oh yeah, I only added the info in the commit message:

Use Object.assign instead of spread operator because of this issue: microsoft/TypeScript#10727.

**Comment by robinandeer** [comment]
Oh dammit, now I duplicated some commits... but the changes should still only be about TS
